### ISSUE TITLE
High memory usage on macOS in headless mode. 

### ISSUE THREAD

**Issue by sarkrui** [seleniumbase/SeleniumBase#1916]
So the issue with the same following (chopped) code didn't occur in GUI mode. With `--headless` arg, Google Chrome instance could take up to 10GiB of ram. Hasn't anyone encountered the same? or is there anything wrong with my code? Thank you, very appreciated.

```python
from collections import deque
from datetime import datetime, timedelta
import argparse
import csv
import os
import logging
import random
import signal
import socket
import sys
import time
import datetime
import threading
import requests
import tracemalloc
import pandas as pd
import paramiko
from selenium.common.exceptions import TimeoutException
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException,ElementNotInteractableException
from seleniumbase import BaseCase
from seleniumbase import MasterQA
from jinja2 import Environment, FileSystemLoader

[some code here]

class AmazonFlow(BaseCase):
	[some code here]
    def test_amazon_flow(self):
        try:
            
            # Set Amazon and check location
            self.set_amazon()

            while not self.check_location():
                self.set_amazon()

            # start the heartbeat thread
            # heartbeat_thread.start()

            
            # Continue with price checking indefinitely
            while True:
                # Only proceed if check_prices returns True
                self.check_prices()
                
                time.sleep(10) 
                
                # convert CSV to nice-looking HTML after checking prices
                csv_to_bootstrap_html('price.csv', 'price.html','template.html')  

                # upload the generated HTML to the server
                upload_file_via_ssh()

        except KeyboardInterrupt:
            print("Interrupted by user")
        except Exception as e:
            print(f"An error occurred: {e}")
        finally:
            self.tearDown()

if __name__ == "__main__":
    BaseCase.main(__name__, __file__,"--uc","--incognito")
```

**Comment by mdmintz** [comment]
I'm unable to reproduce your memory issue with regular SeleniumBase. It might be caused by some of the other things that you're trying to do. Eg, you have ``import threading`` in your code, so you might be performing a high-memory operation with that. Note that SeleniumBase has its own system for multithreading tests via ``pytest-xdist`` with the ``-n`` arg (eg. ``pytest -n4`` for 4 parallel processes).

It might also be an issue with ``--uc`` mode (undetected-chromedriver mode), which is a separate Python library that SeleniumBase uses.

**Comment by sarkrui** [comment]
Thanks you for the input, Michael. Simply removing `--uc` (and keeping `--headless`) arg seems have resolved the issue. I will keep you updated when I could reproduce the issue with some vanilla code.

**Comment by LiPingYen** [comment]
I have encountered a similar issue. When running on the m2 device with the` --headless` option, Google Chrome Helper (GPU) gradually consumes more and more memory, eventually causing the computer to freeze. Through testing, I've observed that this behavior persists regardless of whether the `--uc` is enabled or if multiple threads(`-n2`) are being used.

1. with `--headless`
```
pytest test_get_one_tour_photo_only_BC_final_version.py -n2 --dashboard --html=report.html --rs --crumbs --headless —uc 
```
<img width="952" alt="Screenshot 2023-11-19 at 10 39 31 AM copy" src="https://github.com/seleniumbase/SeleniumBase/assets/47491041/2445092c-428f-4476-85a1-10564f02f666">

2. without `--headless`
```
pytest test_get_one_tour_photo_only_BC_final_version.py -n2 --dashboard --html=report.html --rs --crumbs --uc 
```
<img width="948" alt="Screenshot 2023-11-19 at 10 56 11 AM" src="https://github.com/seleniumbase/SeleniumBase/assets/47491041/ea07798c-ab9c-48fa-bef6-92a4c48c26e0">
